packages/kbn-openapi-generator/src/parser/lib/get_components.ts

```diff
@@ -6,12 +6,124 @@
  * Side Public License, v 1.
  */
 
-import type { OpenApiDocument } from '../openapi_types';
+import type {
+  OpenApiDocument,
+  OpenApiComponentsObject,
+  OpenApiSchemasObject,
+} from '../openapi_types';
+import { extractByJsonPointer } from './helpers/extract_by_json_pointer';
+import { findLocalRefs } from './helpers/find_local_refs';
+import { parseRef } from './helpers/parse_ref';
 
-export function getComponents(parsedSchema: OpenApiDocument) {
-  if (parsedSchema.components?.['x-codegen-enabled'] === false) {
+/**
+ * Returns document components.
+ *
+ * It performs topological sorting of component schemas to enable arbitrary
+ * schemas definition order.
+ */
+export function getComponents(document: OpenApiDocument): OpenApiComponentsObject | undefined {
+  if (document.components?.['x-codegen-enabled'] === false) {
     return undefined;
   }
 
-  return parsedSchema.components;
+  if (!document.components) {
+    return;
```

Comment:
nit: Since we explicitly return `undefined` a couple lines above, we could also return it here in the same explicit way for consistency.

Reply:
Made it a bit differently by collapsing two `if` conditions and using `return` clause without explicit `undefined`.